### PR TITLE
feat(log): support full log to download and refactor the logging module code

### PR DESCRIPTION
**Pull request by ysjemmm**
#### What type of PR is this?
type feat

#### What this PR does / why we need it:
support download full log

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

fix 43x branch could not package docker image pipeline

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-common/src/main/java/com/oceanbase/odc/common/unit/BinarySizeUnit.java [hunk 1]
```diff
@@ -49,6 +51,15 @@ public enum BinarySizeUnit {
         }
     }
 
+    public static long convertTo(long originalSize, @NotNull BinarySizeUnit originalUnit,
```
Comment (yhilmare) [inline server/odc-common/src/main/java/com/oceanbase/odc/common/unit/BinarySizeUnit.java:54]: `BinarySizeUnit.GB.of(1).convert(BinarySizeUnit.B)`

which means 1GB to 1024 * 1024 * 1024 B

Comment (ysjemmm) [reply server/odc-common/src/main/java/com/oceanbase/odc/common/unit/BinarySizeUnit.java:54]: OK，I will change

File: script/functions.sh [hunk 2]
```diff
@@ -300,40 +300,46 @@ function maven_build_jar() {
     return 0
 }
 
+function build_module() {
+    local module=$1
+    local setting_xml_path="${ODC_DIR}/builds/settings.xml"
```
Comment (yhilmare) [inline script/functions.sh:305]: there is no file named this `${ODC_DIR}/builds/settings.xml`

Comment (ysjemmm) [reply script/functions.sh]: I have changed scripts that names are build_libs.sh and function.sh

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskDetailRespHist.java [hunk 3]
```diff
@@ -44,4 +44,6 @@ public class ScheduleTaskDetailRespHist {
 
     private String jobGroup;
 
+    private String fullLogDownloadUrl;
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskDetailRespHist.java:47]: same as above

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskDetailResp.java [hunk 4]
```diff
@@ -48,4 +48,6 @@ public class ScheduleTaskDetailResp {
     private Date createTime;
 
     private Date updateTime;
+
+    private String fullLogDownloadUrl;
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskDetailResp.java:52]: same as above

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskOverview.java [hunk 5]
```diff
@@ -46,6 +48,8 @@ public class ScheduleTaskOverview {
      */
     private String jobGroup;
 
+    private String fullLogDownloadUrl;
```
[the remark this answers is not in the document]
Comment (ysjemmm) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskOverview.java:51]: ok

**Comment by ysjemmm** [comment]
see https://github.com/oceanbase/odc/pull/3317

**Pull request closed without merging**